### PR TITLE
apt picks the wrong versions of dependencies when something newer is available

### PR DESCRIPTION
**Pull request by jglick**
Fixes a problem in #9 apparent only after an update appeared on the repository.

@reviewbybees

### CODE REVIEW

File: src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/JavaContainer/Dockerfile [hunk 1]
```diff
@@ -5,11 +5,12 @@
...
 
+ENV JDK_VERSION 8u144-b01-2
 RUN apt-get update && \
     apt-get install --no-install-recommends -y \
         software-properties-common=0.96.24.17 \
-        openjdk-8-jdk=8u144-b01-2 \
+        openjdk-8-jre-headless=$JDK_VERSION \
+        openjdk-8-jdk-headless=$JDK_VERSION \
         curl=7.55.1-1ubuntu2.1 \
-        wget=1.19.1-3ubuntu1.1 \
```
Comment (jglick) [inline src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/JavaContainer/Dockerfile]: Unrelated, but noticed this was already available in `jenkins/sshd`.

File: src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/JavaContainer/Dockerfile [hunk 2]
```diff
@@ -5,11 +5,12 @@
 # cf. SshdContainer/Dockerfile
 FROM jenkins/sshd:ff2527313f5c
 
+ENV JDK_VERSION 8u144-b01-2
 RUN apt-get update && \
     apt-get install --no-install-recommends -y \
         software-properties-common=0.96.24.17 \
-        openjdk-8-jdk=8u144-b01-2 \
+        openjdk-8-jre-headless=$JDK_VERSION \
+        openjdk-8-jdk-headless=$JDK_VERSION \
```
Comment (jglick) [inline src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/JavaContainer/Dockerfile]: To be really safe I would need to specify every transitive dependency, but there are dozens of them and this seems impractical for now.

**Comment by jglick** [comment]
Hmm, [still seeing problems](https://ci.jenkins.io/job/Core/job/acceptance-test-harness/job/PR-375/2/testReport/plugins/SshSlavesPluginTest/unableToConnectWrongCredentials/) though this job apparently is not running the JUnit Attachments plugin so I am not sure what the failure was.

**Comment by jglick** [comment]
Managed to reproduce a failure in ATH locally, Looking into what is going on.

**Comment by jglick** [comment]
Ah right, just forgot to update parent hashes, will fix.

**Comment by jglick** [comment]
Hmm, [this failure](https://ci.jenkins.io/job/Core/job/acceptance-test-harness/job/master/49/testReport/plugins/XvncPluginTest/workflow/attachments/docker-XvncSlaveContainer.build.log) suggests that even the `=` trick is not foolproof:

```
Step 3/3 : RUN apt-get update &&     apt-get install --no-install-recommends -y         software-properties-common=0.96.24.17         openjdk-8-jre-headless=$JDK_VERSION         openjdk-8-jdk-headless=$JDK_VERSION         curl=7.55.1-1ubuntu2.1         ant=1.9.9-4         maven=3.5.0-6
 ---> Running in 16c1a5558583
Hit:1 http://archive.ubuntu.com/ubuntu artful InRelease
Get:2 http://security.ubuntu.com/ubuntu artful-security InRelease [78.6 kB]
Get:3 http://archive.ubuntu.com/ubuntu artful-updates InRelease [78.6 kB]
Get:4 http://archive.ubuntu.com/ubuntu artful-backports InRelease [72.2 kB]
Get:5 http://archive.ubuntu.com/ubuntu artful-updates/main amd64 Packages [105 kB]
Get:6 http://archive.ubuntu.com/ubuntu artful-updates/universe amd64 Packages [36.9 kB]
Fetched 371 kB in 0s (454 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
E: Version '7.55.1-1ubuntu2.1' for 'curl' was not found
The command '/bin/sh -c apt-get update &&     apt-get install --no-install-recommends -y         software-properties-common=0.96.24.17         openjdk-8-jre-headless=$JDK_VERSION         openjdk-8-jdk-headless=$JDK_VERSION         curl=7.55.1-1ubuntu2.1         ant=1.9.9-4         maven=3.5.0-6' returned a non-zero code: 100
```

**Comment by dwnusbaum** [comment]
I think that `7.55.1-1ubuntu2.1` was removed from distribution, see #11.

**Pull request merged**